File: arduinobot_ws/src/arduinobot_test/script/simple_service_client.py
#!/usr/bin/env python3
import rospy
from arduinobot_test.srv import AddTwoInts
import sys

if __name__ == '__main__':
	if len(sys.argv) == 3:
		a = int(sys.argv[1])
		b = int(sys.argv[2])
	else:
		raise ValueError("Wrong number of requested args")
		sys.exit(-1)

	print(f'Requesting sum of {a}, {b}')

	rospy.wait_for_service("add_two_ints")
	add_two_ints = rospy.ServiceProxy("add_two_ints", AddTwoInts)
	response = add_two_ints(a, b)

	print(f"Service Response: {response}")

File: arduinobot_ws/src/arduinobot_test/src/simple_service_client.cpp
#include "ros/ros.h"
#include "arduinobot_test/AddTwoInts.h"

int main(int argc, char **argv)
{
	ros::init(argc, argv, "add_two_ints_client");

	if(argc != 3)
		{
			ROS_INFO("Wrong number of requested args");
			return -1;
		}

	ros::NodeHandle n;
	ros::ServiceClient client = n.serviceClient<arduinobot_test::AddTwoInts>("add_two_ints");
	arduinobot_test:: AddTwoInts srv;
	srv.request.a = atoll(argv[1]);
	srv.request.b = atoll(argv[2]);
	ROS_INFO("Requesting %d + %d", (int)srv.request.a, (int)srv.request.b);

	if (client.call(srv))
	{
		ROS_INFO("Service Response %d", (int)srv.response.sum);
	}
	else
	{
		ROS_ERROR("Failed to call service add_two_ints");
	}
	return 0;
}